cli/src/config-schema.json

```diff
@@ -168,6 +168,16 @@
                     "type": "string",
                     "enum": ["none", "watchman"],
                     "description": "Whether to use an external filesystem monitor, useful for large repos"
+                },
+                "watchman": {
+                    "type": "object",
+                    "properties": {
+                      "register_trigger": {
+                        "type": "boolean",
+                        "default": false,
+                        "description": "Whether to use triggers to monitor for changes in the background."
```

Comment:
nit: `register_trigger` sounds rather generic. Maybe better to say something like `register_snapshot_trigger` or `auto_snapshot`?

Reply:
given that we are already in the watchman namespace, this seems fine to me as is.  If folks can agree on something here I am happy to change it, but I don't feel like any of these are more compelling than the current name

Reply:
My point is it's unclear what will be triggered. Snapshot is technically just one of the actions that can be triggered on file changes, right?

I agree about the namespace thingy.

Reply:
I find that vaguely convincing.  `register_snapshot_trigger` works for me.